compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/Types.java

```diff
@@ -0,0 +1,226 @@
+        for (BPackageSymbol packageSymbol : imports) {
+            SymbolEnv pkgEnv = symbolTable.pkgEnvMap.get(packageSymbol);
+            BSymbol bSymbol = symbolResolver.lookupSymbolInMainSpace(pkgEnv, typeDef);
+            if (bSymbol != symbolTable.notFoundSymbol) {
+                return bSymbol;
+            }
+        }
+
+        return symbolTable.notFoundSymbol;
+    }
+
+    private Optional<Map<String, Symbol>> getTypeDefSymbolsInModule(PackageID packageID) {
+        BPackageSymbol packageSymbol = packageCache.getSymbol(packageID);
+        if (packageSymbol == null) {
+            return Optional.empty();
+        }
+
+        SymbolEnv pkgEnv = symbolTable.pkgEnvMap.get(packageSymbol);
+        Scope pkgEnvScope = pkgEnv.scope;
+        if (pkgEnvScope != null && pkgEnvScope.entries != null) {
+            Map<String, Symbol> typeDefSymbols = new HashMap<>();
+            for (Scope.ScopeEntry scopeEntry : pkgEnvScope.entries.values()) {
+                BSymbol bSymbol = scopeEntry.symbol;
+                if (isValidTypeDef(bSymbol)) {
+                    String typeDefName = bSymbol.getName().getValue();
+                    typeDefSymbols.put(typeDefName, symbolFactory.getBCompiledSymbol(bSymbol, typeDefName));
+                }
+            }
+
+            for (BPackageSymbol importedPkgs : packageSymbol.imports) {
+                findTypeDefsInPackageSymbol(importedPkgs, typeDefSymbols);
+            }
+
+            return Optional.of(typeDefSymbols);
+        }
+
+        return Optional.empty();
+    }
+
+    private void findTypeDefsInPackageSymbol(BSymbol packageSymbol, Map<String, Symbol> typeDefSymbols) {
+        if (packageSymbol != null) {
+            SymbolEnv pkgEnv = symbolTable.pkgEnvMap.get(packageSymbol);
+            Scope pkgEnvScope = pkgEnv.scope;
+            if (pkgEnvScope != null && pkgEnvScope.entries != null) {
+                for (Scope.ScopeEntry scopeEntry : pkgEnvScope.entries.values()) {
+                    BSymbol bSymbol = scopeEntry.symbol;
+                    if (isValidTypeDef(bSymbol)) {
+                        String typeDefName = packageSymbol.getName().getValue() + ":" + bSymbol.getName().getValue();
+                        typeDefSymbols.put(typeDefName, symbolFactory.getBCompiledSymbol(bSymbol, typeDefName));
+                    } else if (bSymbol.tag == SymTag.PACKAGE) {
+                        findTypeDefsInPackageSymbol(bSymbol, typeDefSymbols);
+                    }
+                }
+            }
+        }
+
+    }
+
+    private boolean isValidTypeDef(BSymbol bSymbol) {
+        return bSymbol != null && bSymbol != symbolTable.notFoundSymbol && bSymbol.getOrigin() != SymbolOrigin.VIRTUAL
+                && (bSymbol.tag == SymTag.TYPE_DEF
+                || bSymbol.tag == SymTag.CONSTANT
+                || bSymbol.tag == SymTag.ENUM
+                || Flags.unMask(bSymbol.flags).contains(Flag.CLASS));
```

Comment:
Same comment here. Additonally, this is not the way to check for flags. Use `Symbols.isFlagOn()`